### PR TITLE
Add CAMI amber uitltiy script

### PR DESCRIPTION
**Pull request by SantaMcCloud**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

File: tools/cami_amber/macros.xml [hunk 1]
```diff
@@ -5,13 +5,32 @@
             <yield/>
         </requirements>
     </xml>
+    <xml name="requirements_script">
+        <requirements>
+            <requirement type="package" version="@SCRIPT_VERSION@">biobox_add_taxid</requirement>
```
Comment (bgruening) [inline tools/cami_amber/macros.xml:10]: If this is only used in one tool, include it there.

File: tools/cami_amber/biobox_add_taxid.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
+<tool id="biobox_add_taxid" name="Biobox add taxid" version="@SCRIPT_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description>Add taxid output from BAT or GTDB(with help of other tools) to binning file(biobox format)</description>
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:2]:
```suggestion
    <description>Add taxid output from BAT or GTDB to biobox binning data</description>
```

Mabye? Try to be conceise.

File: tools/cami_amber/biobox_add_taxid.xml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    <expand macro="requirements_script"/>
+    <command detect_errors="exit_code">
+    <![CDATA[
+
+        mkdir -p input &&
+
+        ln -s '${biobox_file}' '$biobox_file.element_identifier' &&
+
+        #for $f in $input:
+            ln -s '$f' 'input/$f.element_identifier' &&
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:15]: do you need the element identifier? Can you use a random filename?

The problem can be that the element_identifier might not be unique, depending on how the user names the file.

You could do `#for $i, $f in enumerate($input):`

To get something unique.

File: tools/cami_amber/biobox_add_taxid.xml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        #end for
+
+        #if $tool_type.is_select == 'GTDB':
+            ln -s '${gtdb_to_taxdump}' '$gtdb_to_taxdump.element_identifier' &&
+            ln -s '${taxonkit}' '$taxonkit.element_identifier' &&
+            #set $output = $biobox_file.element_identifier.split('.')[0] + "_add_taxid_GTDB." + $biobox_file.element_identifier.split('.')[-1]
+        #end if
+
+        #if $tool_type.is_select == 'BAT':
+            #set $output = $biobox_file.element_identifier.split('.')[0] + "_add_taxid_BAT." + $biobox_file.element_identifier.split('.')[-1]
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:25]: instead of trying to construct the exact file name ... which might brake in the next version, can you maybe do something like that in the end.

`cp *_add_taxid_BAT*` $file.

File: tools/cami_amber/biobox_add_taxid.xml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        <conditional name="tool_type">
+            <param name="is_select" type="select" label="Select the tool which output should be used here" help="Select BAT when you use the bin2classifier file(s) or select GTDB when using the summary file(s)">
+                <option value="BAT">BAT</option>
+                <option value="GTDB">GTDB</option>
+            </param>
+            <when value="GTDB">
+                <param argument="--gtdb_to_taxdump" type="data" format="tabular" label="Input the output from gtdb_to_taxdump here" help="Use the output from gtdb_to_taxdump here since we need the mapped names from GTDB to NCBI to get the arcoding taxids from NCBI"/>
+                <param argument="--taxonkit" type="data" format="tabular" label="Input the output from Taxonkit here" help="Use the output from Taxonkit here since the need the mapped NCBI names to the arcording taxids"/>
+                <param argument="--column" type="integer" min="0" value="0" label="Input the colum with the NCBI names"/>
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:54]: try using the `data_column` type. https://docs.galaxyproject.org/en/latest/dev/schema.html

File: tools/cami_amber/biobox_add_taxid.xml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            </when>
+            <when value="BAT"/>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="file" format="tabular" label="${tool.name}: BIOBOX ADD TAXID COLUMN"/>
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:60]: `file` is imho not a good variable name

File: tools/cami_amber/macros.xml (hunk 1 above)
Comment (SantaMcCloud) [reply tools/cami_amber/macros.xml]: https://github.com/galaxyproject/tools-iuc/pull/6268/commits/e4816c9f4d17c17fbe575c5e06c7f940c4830e5f

File: tools/cami_amber/biobox_add_taxid.xml (hunk 3 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: https://github.com/galaxyproject/tools-iuc/pull/6268/commits/642062bf29edc8af5bfba8eb0cd08066ca5f771b

File: tools/cami_amber/biobox_add_taxid.xml (hunk 4 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: https://github.com/galaxyproject/tools-iuc/pull/6268/commits/5d79ecfc699921e00547c9c00be57480322aa600

File: tools/cami_amber/biobox_add_taxid.xml (hunk 6 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: https://github.com/galaxyproject/tools-iuc/pull/6268/commits/85538235e194828741b88fef51403a9845dac857

File: tools/cami_amber/biobox_add_taxid.xml (hunk 5 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: Yea, I know I had to change it since I wrote my script from different. My input of the column started by 0 not like here by one, but I did change it and release a new version of it to now be able to change it here [https://github.com/galaxyproject/tools-iuc/pull/6268/commits/daa9fe4118aef57dfaa0eae3b2ea1ecc82c45551]. 

I just did it like this to open the PR to be able to fix all other problems till my recipe is updated [https://github.com/bioconda/bioconda-recipes/pull/50287/]

I also will restart the test after the update is merged and the package with the new version is able to be used!

File: tools/cami_amber/biobox_add_taxid.xml [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    <command detect_errors="exit_code">
+    <![CDATA[
+
+        mkdir -p input &&
+
+        ln -s '${biobox_file}' '$biobox_file.element_identifier' &&
+
+        #for $i, $f in enumerate($input):
+            #set $filename = str($i) + ".tsv"
+            ln -s '$f' 'input/$filename' &&
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:19]:
```suggestion
            ln -s '$f' 'input/${i}.tsv' &&
```

File: tools/cami_amber/biobox_add_taxid.xml [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+        #end for
+
+        #if $tool_type.is_select == 'GTDB':
+            ln -s '${gtdb_to_taxdump}' '$gtdb_to_taxdump.element_identifier' &&
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:23]: why is this needed? the symlinking of both path?

File: tools/cami_amber/biobox_add_taxid.xml [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+            cp *_add_taxid_GTDB* $output
+        #else:
+            cp *_add_taxid_BAT* $output
+        #end if
+
+    ]]>
+    </command>
+    <inputs>
+        <param name="biobox_file" type="data" format="tabular" label="CAMI amber biobox file input" help="Input the CAMI amber biobox file here which are corespond with the biner and the BAT/GTDB output which did used the biner as input! "/>
+        <param name="input" type="data" multiple="true" format="tabular" label="Input BAT(bin2classifier) file(s) or GTDB(Summary) file(s) here" help="Input the BAT or GTDB output here which are corespoding with the CAMI amber biobox file(the should use the same biner output as input)"/>
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:48]: If you ask in the select box below if the input is BAT or GTDB why not move this input into the conditional?

File: tools/cami_amber/biobox_add_taxid.xml [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+            ln -s '${taxonkit}' '$taxonkit.element_identifier' &&
+        #end if
+
+        biobox_add_taxid.py
+        '$biobox_file.element_identifier'
+        '$tool_type.is_select'
+        'input'
+        #if $tool_type.is_select == 'GTDB':
+            -g '$gtdb_to_taxdump.element_identifier'
+            -t '$taxonkit.element_identifier'
```
Comment (bgruening) [inline tools/cami_amber/biobox_add_taxid.xml:33]:
```suggestion
            -g '$gtdb_to_taxdump'
            -t '$taxonkit'
```

Why is this not working?

File: tools/cami_amber/biobox_add_taxid.xml (hunk 7 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: https://github.com/galaxyproject/tools-iuc/pull/6268/commits/6f4fb20d97ad02ae3c5f45c0fba89dddb9a216db

File: tools/cami_amber/biobox_add_taxid.xml (hunk 8 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: It is not needed, I just forgot to delete it, since I did keep it for looking for potential bugs/mistakes. I did remove it here https://github.com/galaxyproject/tools-iuc/pull/6268/commits/bb3ad1200300e2e96d88dd2d73254fbeba41178e

File: tools/cami_amber/biobox_add_taxid.xml (hunk 10 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: https://github.com/galaxyproject/tools-iuc/pull/6268/commits/bb3ad1200300e2e96d88dd2d73254fbeba41178e

File: tools/cami_amber/biobox_add_taxid.xml (hunk 9 above)
Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml:42]: I can change it, but in the end both are using the same source. I ask it here since GTDB required more inputs than BAT since BAT gives you directly the Taxids while for GTDB you need 2 other tools. Because of this, used one input for either one of them just that the user has to either upload all files from BAT which he needs for the binning file or the user has to upload all GTDB files which he needs for the binning file. I hope this clear it up a bit, if not I can try to explain differently!

Comment (bgruening) [reply tools/cami_amber/biobox_add_taxid.xml:42]: Thanks for the explanation. I just thought, since we are asking the user eitherway which tool they have used, we can move this data param also into the conditional and make the tool form easier to understand.

For example the help text of this param here would be more clear ... since it will only refer to one input - GTDB OR BAT ... the choice the user did before.

Comment (SantaMcCloud) [reply tools/cami_amber/biobox_add_taxid.xml]: Yes, this make more sense to have a better interface in this way. I did change it, and it should be more clear now, I think? https://github.com/galaxyproject/tools-iuc/pull/6268/commits/6afdca1984e205b3f5f8d160ecb0c280db9e2448 https://github.com/galaxyproject/tools-iuc/pull/6268/commits/29e2e33dc935eb10963c0356ae3af6c9573f069a

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
thanks a lot!

**Comment by SantaMcCloud** [comment]
can you also look at this https://github.com/galaxyproject/tools-iuc/pull/6148#issuecomment-2290893148 @bgruening since Paul is on Variation?
